### ISSUE TITLE
feat: Add live playing websocket

### ISSUE THREAD

**Issue by kyleyannelli** [kyleyannelli/KMusicBot#144]
- Only 1 websocket can be open per user
### Not sure we can use the existing cookies if we just open a websocket, maybe on the initial req before it upgrades?
- Otherwise have to make an HTTP route to get a token that allows opening a websocket as the user. Eg `/api/secure/open-socket` returns some `token`. Open websocket at `/api/secure/websocket`, send the `token` then the connection will continue, otherwise it will close and the token will invalidate after 5 seconds.
- User can send messages of guild id only, websocket will respond with something like
```json
{
    "title": "Time",
    "author": "Pink Floyd",
    "position": 133,
    "length": 413,
    "thumbnail": "https://img.youtube.com/vi/Qr0-7Ds79zo/hqdefault.jpg",
    "listeners": {
        ...
    }
}
```
if nothing is playing we could just
```json
{
    "length": -1
}
```

**Comment by kyleyannelli** [comment]
f7c545df481cd899eb6ef8cb9d832cfb961e71d7 begins work on the websocket. It is fully functional. With testing I can confirm it's working well. 

## Current behavior / system design

Bot --socket--> Spark API Backend:

- Only one connection allowed
- Authenticated via bearer token
- API Backend updated with every guilds NowPlayingResponse every 250ms (overkill?)
- I chose using a websocket rather than something like MQTT or Redis pub sub because it feels like it would bloat the project for how straight forward the now playing information is. Additionally does not exactly fall in line with the current microservices arch of this whole project. A websocket allows for easily secured cross network communication. I think if I rewrote or began writing of this project today, I would have picked a monolithic design. However, I don't think it's right to begin changing the spirit of the project since it has been very easy for me to maintain and upgrade despite not working on it frequently.
- Read Only. The bot tells the API Backend the data. The API Backend messages are simply ignored.

User <--socket-- Spark API:

- User uses existing cookies to authenticate the connection and authorize which guilds they will receive
- User receives NowPlayingResponses for guilds they are in every 1s (because songs meaningfully update at this rate, eg 2 minutes 2 seconds to 2 min 3 seconds)
- I am not a fan of the connection never dying until the user says so. Ex. user gets kicked from guild but they have an open websocket connection, they will still get updates for the guild. The socket should probably close every so often with a reauthenticate message & code so we cannot have dangling connections. I think the load on the server would be too great if we checked the cookies before sending every message. This is because they have to decrypt and additionally would make far too many calls to the discord API assuming there's more than a couple active users on the socket. 
- Read Only. The API Socket tells the user what data the guilds have. Any messages from the user are simply ignored.

**Comment by kyleyannelli** [comment]
Fixed and fully implemented by ed1c6dd60c65a315589b54fc1b91c49463bd2235
>
>     * I am not a fan of the connection never dying until the user says so. Ex. user gets kicked from guild but they have an open websocket connection, they will still get updates for the guild. The socket should probably close every so often with a reauthenticate message & code so we cannot have dangling connections. I think the load on the server would be too great if we checked the cookies before sending every message. This is because they have to decrypt and additionally would make far too many calls to the discord API assuming there's more than a couple active users on the socket.
